### ISSUE TITLE
Consider OAuth2TokenVerifier.verify return object instead of throwing Exception

### ISSUE THREAD

**Issue by rwinch** [jzheaux/spring-security-oauth2-resource-server#24]
Throwing an `OAuth2AuthenticationException` is probably better left to the `AuthenticationProvider`. This API is probably better off returning a response object rather than throwing the exception itself.

**Comment by jzheaux** [comment]
`OAuth2Error verify(T token)` sounds reasonable to me, or are you of the mind to go broader like `OAuth2VerifierResult verify(T token)`?

**Comment by rwinch** [comment]
I was trying to be too prescriptive, but I was also thinking `OAuth2Error`

**Comment by jgrandja** [comment]
Agreed it's best to avoid wasting resources by throwing an exception.

However, `OAuth2Error verify(T token)` wouldn't work. What if there is no error? Does the method return `null`?

`OAuth2TokenVerificationResult verify(T token)` makes more sense for both pass/fail scenarios.
